src/Controller/MessageController.php

```diff
@@ -23,35 +24,45 @@ class MessageController extends AbstractController
     /**
      * TODO: cover this method with tests, and refactor the code (including other files that need to be refactored)
      */
-    #[Route('/messages')]
+    #[Route('/messages', name: 'app_message_list', methods: ['GET'])]
     public function list(Request $request, MessageRepository $messages): Response
     {
-        $messages = $messages->by($request);
-  
-        foreach ($messages as $key=>$message) {
-            $messages[$key] = [
-                'uuid' => $message->getUuid(),
-                'text' => $message->getText(),
-                'status' => $message->getStatus(),
-            ];
-        }
-        
-        return new Response(json_encode([
-            'messages' => $messages,
-        ], JSON_THROW_ON_ERROR), headers: ['Content-Type' => 'application/json']);
+        $messages = $messages->by($request->query->all());
+        $data = array_map($this->transformMessageToArray(...), $messages);
+
+        return $this->json(['messages' => $data], Response::HTTP_OK);
     }
 
-    #[Route('/messages/send', methods: ['GET'])]
+    #[Route('/messages/send', name: 'app_message_send', methods: ['GET'])]
     public function send(Request $request, MessageBusInterface $bus): Response
     {
-        $text = $request->query->get('text');
-        
-        if (!$text) {
+        $text = $request->query->getString('text');
+
+        if ('' === $text) {
             return new Response('Text is required', 400);
         }
+        if (strlen($text) > 255) {
+            return new Response('Text is too long', 400);
+        }
 
         $bus->dispatch(new SendMessage($text));
-        
+
         return new Response('Successfully sent', 204);
     }
-}
\ No newline at end of file
+
+    /**
```

Comment:
Isolated method so it can be easier to upgrade in future. This example is very small, but nested objects are common use-case.